packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -249,6 +249,32 @@ export class ChildLogger extends TelemetryLogger {
         namespace?: string,
         properties?: ITelemetryProperties,
         propertyGetters?: ITelemetryPropertyGetters): TelemetryLogger {
+        // if we are creating a child of a child, rather than nest, which will increase
+        // the callstack overhead, just generate a new logger that includes everything from the previous
+        if (baseLogger instanceof ChildLogger) {
+            const combinedProperties =
+                baseLogger.properties === undefined && properties === undefined
+                    ? undefined
+                    : {
+                        ...baseLogger.properties,
+                        ... properties,
```

Comment:
```suggestion
                        ...properties,
```

Reply:
if we care about this there should be a lint rule